### DISCUSSION TITLE
Alternative to Regexp in routes

### DISCUSSION THREAD

**Discussion by apalumbo** [remix-run/remix#4472] [Q&A] [answered]
We would like to migrate our e-commerce solution from NextJS to Remix , we are stuck to an issue on routing as our rewrites use a suffix to match routes like 
/en/:product_handle(.*)-p 
/en/:collection_handle(.*)-c

In this way we have an url that is good for SEO and everything is optimised for each route.

I understood that the underline router of remix has removed the support to regexp , I am wondering if there are other ways to support this use case, I tried also defining the rewrites in the remix config in routes but the capabilities are the same

Thank you

**Comment by kiliman** [comment, marked as answer]
I would just create a splat `$.tsx` route, then parse the path and handle accordingly.
```ts
export const loader = async ({request, params}: LoaderArgs) => {
  const splat = params['*']
  if (splat.endsWith('-p')) {
    // handle product
  else if (splt.endsWith('-c')) {
    // handle collection
  } else {
    // handle default
  }
}
```

**Reply by apalumbo** [reply]
Thank you for your suggestion @kiliman ,  this will also bring the same `if` chain in the page component ? Is this the only way, I am thinking to the implication on bundling (I am a rookie on remix and I am not aware on how it splits the bundle for client components)

**Reply by kiliman** [reply]
This is in the loader, which runs only on the server, so won't have any effect on the bundle. However your UI components will need to know how to deal with the loader data.

You could use lazy loading if you need to split the components, but unless your route is really heavy, I wouldn't worry about it.
